src/node_crypto.cc

```diff
@@ -6975,30 +6975,17 @@ void TimingSafeEqual(const FunctionCallbackInfo<Value>& args) {
 }
 
 void InitCryptoOnce() {
-  SSL_load_error_strings();
-  OPENSSL_no_config();
+  OPENSSL_INIT_SETTINGS* settings = OPENSSL_INIT_new();
 
   // --openssl-config=...
   if (!per_process::cli_options->openssl_config.empty()) {
-    OPENSSL_load_builtin_modules();
-#ifndef OPENSSL_NO_ENGINE
-    ENGINE_load_builtin_engines();
-#endif
-    ERR_clear_error();
-    CONF_modules_load_file(per_process::cli_options->openssl_config.c_str(),
-                           nullptr,
-                           CONF_MFLAGS_DEFAULT_SECTION);
-    int err = ERR_get_error();
-    if (0 != err) {
-      fprintf(stderr,
-              "openssl config failed: %s\n",
-              ERR_error_string(err, nullptr));
-      CHECK_NE(err, 0);
-    }
+    const char* conf = per_process::cli_options->openssl_config.c_str();
+    OPENSSL_INIT_set_config_filename(settings, conf);
   }
 
-  SSL_library_init();
-  OpenSSL_add_all_algorithms();
+  OPENSSL_init_ssl(0, settings);
+  OPENSSL_INIT_free(settings);
+  settings = NULL;
```

Comment:
```suggestion
  settings = nullptr;
```

To fix [linter](https://travis-ci.com/nodejs/node/jobs/246485610#L308-L311) error:
```console
Running C++ linter...

src/node_crypto.cc:6988:  Use nullptr instead of NULL  [readability/null_usage] [2]

Done processing src/node_crypto.cc

Total errors found: 1
```

Reply:
That's weird, how come my local `make test` didn't catch that?

Reply:
hm, it did, but it didn't stop the build from progressing, and got covered by the passing tests afterwards.

Reply:
Ah it's a regression. I'll open a PR.

Reply:
https://github.com/nodejs/node/pull/30012 to make lint failures fail.